daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -51,9 +51,36 @@ final case class VersionedTransaction[Nid, +Cid](
       version = latestWhenAllPresent(version, languageVersions map (a => a: SpecifiedVersion): _*),
     )
   }
+
+  def nodes: HashMap[Nid, GenNode.WithTxValue[Nid, Cid]] =
+    transaction.nodes
+
+  def roots: ImmArray[Nid] =
+    transaction.roots
+
+  def foreach(f: (Nid, GenNode.WithTxValue[Nid, Cid]) => Unit): Unit =
+    transaction.foreach(f)
+
+  def fold[A](z: A)(f: (A, (Nid, GenNode.WithTxValue[Nid, Cid])) => A): A =
+    transaction.fold[A](z)(f)
+
+  def foldWithPathState[A, B](globalState0: A, pathState0: B)(
+      op: (A, B, Nid, GenNode.WithTxValue[Nid, Cid]) => (A, B)): A =
+    transaction.foldWithPathState[A, B](globalState0, pathState0)(op)
+
+  def localContracts[Cid2 >: Cid]: Map[Cid2, Nid] =
+    transaction.localContracts
+
+  def inputContracts[Cid2 >: Cid]: Set[Cid2] =
+    transaction.inputContracts
```

Comment:
`transaction` is already a public member. If you want to keep it that way, I would rather keep those methods where they are and not redefining them here.

Reply:
You are right. 
The problem is that I do not want code external to LF relies on `GenTransaction` anymore.
This code should not try to build a transaction, but relies only on the Engine or the protobuf Decoder to produce transaction.

For testing,  one should use the `TransactionBuilder` I refactor in #6294

Reply:
This looks getter-lensy, so suggestion in branch `remy+s11-control-lf-versions`, specifically 66528f2e4ba0469961f4fb9098de7025d1bda974.

https://github.com/digital-asset/daml/blob/66528f2e4ba0469961f4fb9098de7025d1bda974/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala#L288-L291

Reply:
:+1: